src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/ServiceConnectionsController.php

```diff
@@ -52,6 +54,22 @@ public function __invoke(): RedirectResponse|Response
         );
     }
 
+    #[IsGranted("ROLE_SURFCONEXT_REPRESENTATIVE")]
+    #[Route(
+        path: '/connections-download',
+        name: 'service_connections_download',
+        methods: ['GET']
+    )]
+    public function download(): Response
+    {
+        $entities = $this->serviceConnectionService->getExportData($this->getInstitutionId());
```

Comment:
IMHO controllers should not do anything. Their job is to delegate this to "specialists" and return the result. In thios case it is so small, but we've seen huge controllers in this domain. Just to keep in mind.

Reply:
SPD typically has very slimm controllers. Save for some of the more involved form heavy actions. I feel we are well balanced here.